src/CoxeterMatrices.jl

```diff
@@ -0,0 +1,216 @@
+    if C == 'A'
+        if !isAffine
+            M = fill(2,N,N)
+            for i = 1:N
+                M[i,i] = 1
+                if i != N
+                    M[i,i+1] = 3
+                    M[i+1,i] = 3
+                end
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('A',N+1,false)
+            M[1,N+1] = 3
+            M[N+1,1] = 3
+        end
+    elseif C =='B' || C == 'C'
+        if !isAffine
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            if N>1
+                M[N-1,N] =4
+                M[N,N-1]=4
+            else
+                error("B and C types must be of size >=2")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('B',N+1,false)
+            if C == 'B'
+                M[1,2]=2
+                M[2,1]=2
+                M[1,3]=3
+                M[3,1]=3
+            else
+                M[1,2]=4
+                M[2,1]=4
+            end
+
+        end
+    elseif C == 'D'
+        if !isAffine 
+            if N<4 
+                error("D types must be of size >=4")
+            end
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            M[N-1,N]=2
+            M[N,N-1]=2
+            M[N-2,N]=3
+            M[N,N-2]=3
+        else
+            M = build_coxeter_matrix_from_group_type('D',N+1,false)
+            M[1,2]=2
+            M[2,1]=2
+            M[1,3]=3
+            M[3,1]=3
+        end
+    elseif C == 'E'
+        if !isAffine
+            M = build_coxeter_matrix_from_group_type('A',N,false)
+            if N == 6
+                M[5,6]=2
+                M[6,5]=2
+                M[3,6]=3
+                M[6,3]=3
+            elseif N ==7
+                M[6,7]=2
+                M[7,6]=2
+                M[3,7]=3
+                M[7,3]=3
+            elseif N == 8
+                M[7,8]=2
+                M[8,7]=2
+                M[5,8]=3
+                M[8,5]=3
+            else
+                error("E types must be of size 6,7 or 8")
+            end
+        else
+            M = build_coxeter_matrix_from_group_type('A',N+1,false)
+            if N == 6
+                M[5,6]=2
+                M[6,5]=2
+                M[3,6]=3
+                M[6,3]=3
+            elseif N == 7
+                M[7,8]=2
+                M[8,7]=2
+                M[4,8]=3
+                M[8,4]=3
+            elseif N == 8
+                M[8,9]=2
+                M[9,8]=2
+                M[6,9]=3
+                M[9,6]=3
+            else
+                error("E~ types must be of size 6,7 or 8")
+            end
+        end
+    elseif C == 'F'
+        if !isAffine
+            if N < 4
+                M = build_coxeter_matrix_from_group_type('A',N,false)
+                M[N-2,N-1] = 4
+                M[N-1,N-2] = 4
+            else
+                error("F and F~ types must be of size >=4")
```

Comment:
Type `F` should only have `F4`: anything higher in this family is not a finite group.